crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -930,9 +1035,15 @@ impl CheckpointBuilder {
                 [(sequence_number, summary)],
             )?;
         }
+
+        self.state
+            .get_cache_commit()
+            .commit_transactions(&all_tx_digests)
```

Comment:
probably don't need this - transactions should be in `pending_execution` table

Reply:
also - probably need to delay removal from `pending_execution` until commit time

Reply:
Turns out this is not so simple, as checkpoints may include locally executed fastpath transactions that may not be in the pending_execution table.